File: Easy/26-remove-duplicates-from-sorted-array/try-7.cpp
/*
 * Problem #26: Remove Duplicates from Sorted Array
 * Difficulty: Easy
 * Submission: Try 7
 * status: Accepted
 * Language: cpp
 * Date: 21/11/2024, 21:57:14
 * Link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/
 */

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int j=0;
        for (int i = 0; i < nums.size();i++) {
            if (nums[i] != nums[j]){
                nums[++j] = nums[i];    
            }
        }
        return j+1;
    }
};

File: Easy/1-two-sum/try-7.py
/*
 * Problem #1: Two Sum
 * Difficulty: Easy
 * Submission: Try 7
 * status: Accepted
 * Language: python3
 * Date: 23/11/2024, 13:50:57
 * Link: https://leetcode.com/problems/two-sum/
 */

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={}
        for i in range(len(nums)):
            map[nums[i]]=i
        
        for i in range(len(nums)):
            comp=target-nums[i]
            if comp in map and i!=map[comp]:
                return [i,map[comp]]
        return []

        
